### PR TITLE
W-15805907: assign_permission_set_licenses accepts the api_names in either developerName or the PermissionSetLicenseKey

### PR DESCRIPTION
**Pull request by lakshmi2506**
To handle the compatibility issues and ensure smooth functionality for both `get_available_permission_set_licenses` and `assign_permission_set_licenses`, you can make the following adjustments:

1. **Revert `get_available_permission_set_licenses` to Return `PermissionSetLicenseKey`:**
   - This will maintain backward compatibility.

2. **Update `assign_permission_set_licenses` to Accept Both `DeveloperName` and `PermissionSetLicenseKey`:**
   - Modify the function to handle input as either `DeveloperName` or `PermissionSetLicenseKey`, ensuring it can correctly interpret and process both formats. This adjustment will resolve issues with names like `alpha.beta` where the `DeveloperName` is `alpha_beta`.  Accepts api_name as  `alpha_beta` or  `alpha.beta` and assigns them correctly

### CODE REVIEW

**Comment by lakshmi2506** [comment]
Hi @jkasturi-sf @jstvz @vsbharath 

Please review the PR

Thank you

File: cumulusci/tasks/salesforce/users/permsets.py [hunk 1]
```diff
@@ -174,12 +174,34 @@ class AssignPermissionSetLicenses(AssignPermissionSets):
...
     assignment_lookup = "PermissionSetLicenseId"
     assignment_child_relationship = "PermissionSetLicenseAssignments"
 
+    def _get_perm_ids(self):
+        perms_by_ids = {}
+        api_names = "', '".join(self.options["api_names"])
+        missing_perms = {api_name: True for api_name in self.options["api_names"]}
+        for permission_field in self.permission_name_field:
+            perms = self.sf.query(
+                f"SELECT Id,{permission_field} FROM {self.permission_name} WHERE {permission_field} IN ('{api_names}')"
```
Comment (jstvz) [inline cumulusci/tasks/salesforce/users/permsets.py:189]: Have you considered combining these into a single query?

File: cumulusci/tasks/salesforce/users/permsets.py [hunk 2]
```diff
@@ -174,12 +174,34 @@ class AssignPermissionSetLicenses(AssignPermissionSets):
...
+            )
+            perms_by_ids_subset = {
+                p["Id"]: p[permission_field] for p in perms["records"]
```
Comment (jstvz) [inline cumulusci/tasks/salesforce/users/permsets.py:192]: This function ultimately queries both `DeveloperName` and `PermissionSetLicenseKey`, but it ends up potentially clobbering the `DeveloperName` with `PermissionSetLicenseKey`. That's inconsistent, but probably fine if displaying the actual found name is important for debugging.

**Review by jstvz — CHANGES_REQUESTED** [review]
Change requested: Please update the `task_docs` to mention that `api_names` can contain `DeveloperName` or `PermissionSetLicenseKey`. Other comments optional.

**Comment by lakshmi2506** [comment]
Hi @jstvz 

Combined the queries into one, updated the task docs.

Please review the PR
Thank you

**Review by jstvz — APPROVED** [review]

**Pull request merged**